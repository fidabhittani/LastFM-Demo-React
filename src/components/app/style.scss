@import "~styles/variables";

@import "styles/_mixins.scss";

/**
*   MAIN STYLES
*/
.main {
  height: 100%;
  overflow: hidden;

  > header {
    height: 50px;
    color: $textColor;
    background: white;
    box-shadow: 0px 0px 5px rgba($color: $secondaryColor, $alpha: 1);
    display: flex;
    align-items: center;
    padding: 0px 5em;
    justify-content: space-between;
    transition: all 0.2s ease-in-out;

    > h3 {
      color: rgba($color: $primaryColor, $alpha: 1);
    }
    .right-side-content {
      display: flex;
    }

    @include xs {
      padding: 0px 1em;
    }
    @include sm {
      padding: 0px 2em;
    }
    @include md {
      padding: 0px 3em;
    }
    @include xl {
      padding: 0px 5em;
    }
  }
  .content {
    position: relative;
    background: white;
    // background: rgba(100, 100, 100, 0.05);
  }
}
