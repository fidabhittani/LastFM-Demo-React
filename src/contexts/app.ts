import { createContext } from "react";

const AppContext = createContext({});
const { Provider, Consumer } = AppContext;

export const AppProvider = Provider;
export const AppConsumer = Consumer;

export default AppContext;
