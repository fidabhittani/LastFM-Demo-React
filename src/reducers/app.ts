import { SET_LOADING, RESET_LOADING } from "config/constants";
import { IAction, IAppState } from "config/types";

/** APP Initial State Values */
export const initialState = {
  loading: { app: { text: "Loading...", status: false } },
  messages: []
};

const defaultContext = `app`;

export default (state: IAppState = initialState, action: IAction) => {
  const { type, payload } = action;
  switch (type) {
    case SET_LOADING:
      return {
        ...state,
        loading: {
          ...state.loading,
          [payload.context || defaultContext]: { ...payload }
        }
      };

    case RESET_LOADING:
      const { context: resetContext }: any = payload || {};

      return {
        ...state,
        loading: {
          ...state.loading,
          [resetContext || defaultContext]: {
            status: false,
            text: "Loading..."
          }
        }
      };

    default:
      return state;
  }
};
