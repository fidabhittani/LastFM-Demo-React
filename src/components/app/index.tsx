import React, { useState, useEffect, Suspense } from "react";
import i18n from "i18n";
import { useTranslation } from "react-i18next";
import { Switch, Route, Redirect } from "react-router-dom";
import Language from "components/ui/language";
import TopMenus from "components/ui/top.menus";
import { AppProvider } from "contexts/app";
import Loader from "components/ui/loader";
import ArtistContainer from "components/artist";
import EventsContainer from "components/events";

import "./style.scss";
/**
 *  Async Loading of Containers
 */

const Root = () => {
  /** HOOKS : React default hooks declarations and assignments */
  const [language, setLanguage] = useState("en");
  const { t } = useTranslation();

  useEffect(() => {
    i18n.changeLanguage(language);
  }, [language]);

  return (
    <AppProvider value={{ language, setLanguage }}>
      <div className="main">
        <header>
          <h3>{t("title")}</h3>
          <div className="right-side-content">
            <TopMenus />
            <Language />
          </div>
        </header>

        <div className="content">
          <Suspense fallback={<Loader />}>
            <Switch>
              <Route exact path="/artists" render={() => <ArtistContainer />} />
              <Route
                path="/artist/:id/events/"
                render={() => <EventsContainer />}
              />
              <Redirect from="/" to="/artists" />
            </Switch>
          </Suspense>
        </div>
      </div>
    </AppProvider>
  );
};

export default Root;
