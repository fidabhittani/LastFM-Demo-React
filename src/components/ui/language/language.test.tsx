import React from "react";
import { shallow } from "enzyme";
import Language from "./index";
import { AppProvider } from "contexts/app";
/** Language TEST */
describe("Language Root Component Tests", () => {
  const WrappedLanguage = () => (
    <AppProvider
      value={{
        language: "en",
        setLanguage: (lang: string) => console.log(lang)
      }}
    >
      <Language />
    </AppProvider>
  );
  it("Language Component renders correctly", () => {
    const langWrapper = shallow(<WrappedLanguage />);
    expect(langWrapper).toMatchSnapshot();
  });
});
