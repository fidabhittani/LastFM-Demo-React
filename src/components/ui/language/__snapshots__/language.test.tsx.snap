// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language Root Component Tests Language Component renders correctly 1`] = `ShallowWrapper {}`;
